app/services/process_feed_service.rb

```diff
@@ -155,7 +153,8 @@ def status_from_xml(entry)
         reply: thread?(entry),
         language: content_language(entry),
         visibility: visibility_scope(entry),
-        conversation: find_or_create_conversation(entry)
+        conversation: find_or_create_conversation(entry),
+        thread: thread?(entry) ? find_status(thread(entry).first) : nil
```

Comment:
No, do not do this please. First of all, new conversation wouldn't be created in validation callback if conversation is already set (line above). Secondly, the reason this is set in the other place is because of the transaction block and what find_status does.